src/portable_endian.h

```diff
@@ -10,14 +10,33 @@
 
 #endif
 
-// GLIBC / Linux with endian(3) support, which was added in glibc 2.9.
-// Intended to support CentOS 6 and newer.
-#if defined(__linux__) && \
-    ((__GLIBC__ > 3) || \
-     (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 9))
-
+#if defined(HAVE_ENDIAN_H)
```

Comment:
I think I get the idea here...rather than having specific cases for every OS (which admittedly seems to have been the original intent), the new idea is to include either `<endian.h>` or `<sys/endian.h>` as needed (or in some cases neither one), and then add OS-specific definitions as needed.  This looks like a reasonable and somewhat more flexible approach.  But...it looks like this might break the Cygwin case.  Cygwin appears to have `<endian.h>`, but the only way to get to `#elif defined(__CYGWIN__)` is to have neither `<endian.h>` nor `<sys/endian.h>`.  Let me know if I read this wrongly.

Reply:
Moving the `CYGWIN` test to the top.  Will rebuild and try a new binary.

Reply:
Built and installed.  Seems to work, though I'm not sure about all the places where/how endianess is used...